pkg/apis/samples/v1/types.go

```diff
@@ -424,9 +428,23 @@ func (s *Config) ClusterNeedsCreds() bool {
 		s.Spec.ManagementState == operatorv1.Unmanaged {
 		return false
 	}
-	if s.Status.Conditions == nil {
+	if s.Status.Conditions == nil || len(s.Status.Conditions) == 0 {
```

Comment:
the length check is redundant since your for loop below will no-op if the length is zero.

Reply:
update pushed